perfkitbenchmarker/benchmarks/nginx_wrk_benchmark.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run wrk against a simple Nginx web server.
+
+`wrk` is a scalable web load generator.
+`Nginx` is a popular static web server and reverse proxy.
+"""
+
+import functools
+import logging
+import urlparse
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.packages import nginx
+from perfkitbenchmarker.packages import wrk
+
+
+flags.DEFINE_integer('nginx_wrk_test_length', 60,
+                     'Length of time, in seconds, to run wrk at each '
+                     'connction count', lower_bound=1)
+flags.DEFINE_float('nginx_wrk_max_p50_ratio', 5.0,
```

Comment:
I see your point. But I think this would be sensitive to spikes and might cause an early termination of the test

I might be overkill but perhaps something like statistical significance test, or statistical change-point analysis test[1].

http://www.variation.com/cpa/tech/changepoint.html#Change-Point Analysis